services/processor/processor/lib/sync_from_shotgrid.py

```diff
@@ -0,0 +1,282 @@
+    def log(self):
+        if self._log is None:
+            self._log = logging.getLogger(self.__class__.__name__)
+        return self._log
+
+    def _check_shotgrid_project(self):
+        """ Ensure Shotgrid project exists and has all required fields.
+        """
+        self.log.info("Ensuring Shotgrid Project has all the required fields.")
+
+        missing_attrs = get_sg_missing_ay_attributes(self._sg_session)
+
+        if missing_attrs:
+            msg = (f"Shotgrid is missing attributes: {missing_attrs}")
+            self.log.warning(msg)
+            raise ValueError(msg)
+
+    def sync_to_ayon(self, preset_name=None):
+        self.log.info("Started Ayon Syncronization with Shotgrid.")
+        self.log.info(f"Project Name: {self._ay_project['name']}")
+
+        sync_start_time = time.perf_counter()
+        self.log.info("Loading Entities data from Ayon.")
+        self._entity_hub.query_entities_from_server()
+        sync_step_time = time.perf_counter()
+        self.log.info(
+            f"Populating the EntitHub took {sync_step_time - sync_start_time}."
+        )
+
+        self.log.info("Adding Shotgrid Statuses and Tasks to Ayon.")
+        create_sg_entities_in_ay(
+            self._entity_hub.project_entity,
+            self._sg_session,
+            self._sg_project,
+        )
+        prev_step_time = sync_step_time
+        sync_step_time = time.perf_counter()
+        self.log.debug((
+            "Adding Shotgrid Statuses and Tasks to Ayon"
+            f"took {sync_step_time - prev_step_time} to load."
+        ))
+
+        self.log.info("Querying all Shotgrid Entities.")
+        (
+            self._sg_entities_by_id,
+            self._sg_entities_by_parent_id
+        ) = get_sg_entities(
+            self._sg_session,
+            self._sg_project,
+        )
+
+        prev_step_time = sync_step_time
+        sync_step_time = time.perf_counter()
+        self.log.debug((
+            "Querying all Shotgrid Entities and hierrachy took"
+            f"{sync_step_time - prev_step_time} to process."
+        ))
+
+        self._match_shotgrid_hierarchy()
+        self._entity_hub.commit_changes()
+
+        prev_step_time = sync_step_time
+        sync_step_time = time.perf_counter()
+        self.log.debug((
+            "Creating all missing entities in ayon took "
+            f"{sync_step_time - prev_step_time} to process."
+        ))
+
+    def _match_shotgrid_hierarchy(self):
+        project_entity = self._entity_hub.project_entity
+        ayon_project_shotgrid_id_attrib = project_entity.attribs.get_attribute(
+            SHOTGRID_ID_ATTRIB
+        ).value
+
+        if not ayon_project_shotgrid_id_attrib:
+            self.log.error((
+                "Project creation probably went wrong, there's no shotgridID"
+                f"{ayon_project_shotgrid_id_attrib}"
+            ))
+            return
+        elif int(self._sg_project["id"]) != int(ayon_project_shotgrid_id_attrib):
```

Comment:
```suggestion
        if int(self._sg_project["id"]) != int(ayon_project_shotgrid_id_attrib):
```

Reply:
I need this to run only if `ayon_project_shotgrid_id_attrib` is not empty, otherwise the `int` cast would fail.

Reply:
But that handles the previous condition, which ends with `return`. So the `elif` is not needed because, if previous condition happens, it will never get to this line.